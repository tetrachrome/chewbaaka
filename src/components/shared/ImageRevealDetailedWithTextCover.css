/**
 * ImageRevealDetailedWithTextCover.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 05, 2020
 * Updated  : Jul 05, 2020
 */

div.ImageRevealDetailedWithTextCoverComponentCoverDiv {
  width: 120px;
  height: 120px;
  text-align: center;
  vertical-align: middle;
}

div.ImageRevealDetailedWithTextCoverComponentCoverDiv p {
  padding-top: 40px;
  font-size: 16px;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  color: white;
}

div.ImageRevealDetailedWithTextCoverComponentContentDiv {
  width: 120px;
  height: 120px;
  text-align: center;
  vertical-align: middle;
}

div.ImageRevealDetailedWithTextCoverComponentContentDiv img {
  width: 80px;
  height: 80px;
  margin: auto;
}

p.ImageRevealDetailedWithTextCoverComponentDescriptionText {
  font-size: 11px;
}
