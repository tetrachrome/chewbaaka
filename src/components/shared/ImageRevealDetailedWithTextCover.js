/**
 * ImageRevealDetailedWithTextCover.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 05, 2020
 * Updated  : Jul 05, 2020
 */

/**
 * ImageRevealDetailedWithTextCover
 *
 * ImageRevealDetailedWithTextCover is a component that
 * initially shows a block of text specified by the user,
 * and when hovered, hides the text and shows a user
 * specified content image and a description text.
 *
 * Props:
 *  - `title`: The title that is initially shown.
 *
 *  - `contentImage`: The content image to show upon hovered.
 *
 *  - `description`: The content text to show upon hovered.
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Image, Reveal } from 'semantic-ui-react'

import './ImageRevealDetailedWithTextCover.css'

class ImageRevealDetailedWithTextCover extends React.Component {

  render() {
    const defaultColor = "green";

    const componentCoverDivStyles = {
      backgroundColor: (this.props.coverColor ?  this.props.coverColor : defaultColor)
    }

    return (
      <Reveal animated={this.animation()} data-testid="ImageRevealDetailedWithTextCoverComponentTestId">
        <Reveal.Content visible data-testid="ImageRevealDetailedWithTextCoverComponentVisiblePartTestId">
          <div className="ImageRevealDetailedWithTextCoverComponentCoverDiv" style={componentCoverDivStyles}>
            <p>{this.props.title}</p>
          </div>
        </Reveal.Content>
        <Reveal.Content hidden data-testid="ImageRevealDetailedWithTextCoverComponentHiddenPartTestId">
          <div className="ImageRevealDetailedWithTextCoverComponentContentDiv">
            <Image src={this.props.contentImage} />
            <p
              className="ImageRevealDetailedWithTextCoverComponentDescriptionText"
              data-testid="ImageRevealDetailedWithTextCoverComponentDescriptionTextPartTestId"
            >
                {this.props.description}
            </p>
          </div>
        </Reveal.Content>
      </Reveal>
    )
  }

  animation() {
    const supportedAnimations = [
      "fade",
      "small fade",
      "move",
      "move right",
      "move up",
      "move down",
      "rotate",
      "rotate left"
    ];

    let idx = supportedAnimations.indexOf(this.props.animation);

    if ( idx === -1 ) {
      idx = Math.floor(Math.random() * supportedAnimations.length);
    }

    return supportedAnimations[idx];
    // return "fade";
  }
}

export default ImageRevealDetailedWithTextCover;
