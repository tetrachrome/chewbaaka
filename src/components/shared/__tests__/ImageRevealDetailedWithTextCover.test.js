/**
 * ImageReveal.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 04, 2020
 * Updated  : Jul 04, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import ImageRevealDetailedWithTextCover from '../ImageRevealDetailedWithTextCover';

import contentImage from '../assets/cheetah-conservation-fund-30-logo.svg'

test('renders ImageRevealDetailedWithTextCover compoent', () => {
  const title = "Acinonyx jubatus";

  const description = "Cheetah";

  render(
    <ImageRevealDetailedWithTextCover
      title={title}
      contentImage={contentImage}
      description={description}
    />
  );

  const imageRevealComponent = screen.getByTestId('ImageRevealDetailedWithTextCoverComponentTestId');
  expect(imageRevealComponent).toBeInTheDocument();

  // Tests visible part.
  const imageRevealComponentVisiblePart = screen.getByTestId('ImageRevealDetailedWithTextCoverComponentVisiblePartTestId');
  expect(imageRevealComponentVisiblePart).toBeInTheDocument();

  // Tests hidden part.
  const imageRevealComponentHiddenPart = screen.getByTestId('ImageRevealDetailedWithTextCoverComponentHiddenPartTestId');
  expect(imageRevealComponentHiddenPart).toBeInTheDocument();

  // Tests description text part.
  const imageRevealComponentTextDescriptionPart = screen.getByTestId('ImageRevealDetailedWithTextCoverComponentDescriptionTextPartTestId');
  expect(imageRevealComponentTextDescriptionPart).toBeInTheDocument();
  expect(imageRevealComponentTextDescriptionPart.textContent).toBe(description);

  // TODO: test component props.
});
