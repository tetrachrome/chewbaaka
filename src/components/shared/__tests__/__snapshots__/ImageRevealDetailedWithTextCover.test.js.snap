// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageRevealDetailedWithTextCover component snapshot 1`] = `
<div
  className="ui rotate left reveal"
  data-testid="ImageRevealDetailedWithTextCoverComponentTestId"
>
  <div
    className="ui visible content"
    data-testid="ImageRevealDetailedWithTextCoverComponentVisiblePartTestId"
  >
    <div
      className="ImageRevealDetailedWithTextCoverComponentCoverDiv"
      style={
        Object {
          "backgroundColor": "green",
        }
      }
    >
      <p>
        Acinonyx jubatus
      </p>
    </div>
  </div>
  <div
    className="ui hidden content"
    data-testid="ImageRevealDetailedWithTextCoverComponentHiddenPartTestId"
  >
    <div
      className="ImageRevealDetailedWithTextCoverComponentContentDiv"
    >
      <img
        className="ui image"
        src="cheetah-conservation-fund-30-logo.svg"
      />
      <p
        className="ImageRevealDetailedWithTextCoverComponentDescriptionText"
        data-testid="ImageRevealDetailedWithTextCoverComponentDescriptionTextPartTestId"
      >
        Cheetah
      </p>
    </div>
  </div>
</div>
`;
